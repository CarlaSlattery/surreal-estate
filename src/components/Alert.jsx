import React from "react";
// eslint-disable-next-line import/no-extraneous-dependencies
import PropTypes from "prop-types";
import "../styles/alert.css";

const Alert = ({ message, success }) => {
  if (!message) return null;
  return (
    <div className={`alert alert-${success ? "success" : "error"}`}>
      {message}
    </div>
  );
};

Alert.propTypes = {
  message: PropTypes.string.isRequired,
  success: PropTypes.bool,
};

Alert.defaultProps = {
  success: false,
};
export default Alert;
