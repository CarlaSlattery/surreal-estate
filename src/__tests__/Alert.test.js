import React from "react";
import { render, screen } from "@testing-library/react";
import Alert from "../components/Alert";

describe("Alert", () => {
  test("does not render if message is an empty string", () => {
    const { asFragment } = render(<Alert message="" />);

    expect(asFragment()).toMatchSnapshot();
  });

  test("displays error message", () => {
    const { asFragment } = render(
      <Alert message="Error - server not working, please try later." />
    );

    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByText(/error/i)).toBeInTheDocument();
  });
  test("displays success message", () => {
    const { asFragment } = render(<Alert message="Success! Property Added" />);

    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByText(/success/i)).toBeInTheDocument();
  });
});
