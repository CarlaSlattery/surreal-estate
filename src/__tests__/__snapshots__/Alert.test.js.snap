// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert displays error message 1`] = `
<DocumentFragment>
  <div
    class="alert alert-error"
  >
    Error - server not working, please try later.
  </div>
</DocumentFragment>
`;

exports[`Alert displays success message 1`] = `
<DocumentFragment>
  <div
    class="alert alert-error"
  >
    Success! Property Added
  </div>
</DocumentFragment>
`;

exports[`Alert does not render if message is an empty string 1`] = `<DocumentFragment />`;
